### ISSUE TITLE
Example in documention does not work

### ISSUE THREAD

**Issue by vjacobjo** [jesper-raemaekers/python-polarion#101]
Hi,

I was looking into the documents for updating test steps here: https://python-polarion.readthedocs.io/en/latest/workitem.html#test-case-workitem

I use the exact same code copied from the documentation, 
```
wi = project.getWorkitem('ISBUTEST-7351')
wi._polarion_test_steps.steps.TestStep[0].values.Text[1].content = 'New description!'

# getting the test management service and saving the test steps
service = client.getService('TestManagement')
service.setTestSteps(wi.uri, wi._polarion_test_steps.steps.TestStep)

#reload workitem so it's in sync
wi._reloadFromPolarion()
```
I got the following error:

> Exception has occurred: AttributeError
'NoneType' object has no attribute 'steps'
  File "C:\Hawk\ias_desktop_automation\spikes\testing_modify_steps.py", line 10, in <module>
    wi._polarion_test_steps.steps.TestStep[0].values.Text[1].content = 'New description!'
AttributeError: 'NoneType' object has no attribute 'steps'

I did modify the code to retrieve the test steps from the test management service:
```
service = client.getService("TestManagement")

wi = project.getWorkitem('ISBUTEST-7351')

test_steps = service.getTestSteps(wi.uri)
test_steps.steps.TestStep[0].values.Text[1].content = 'New description!'

# getting the test management service and saving the test steps
service = client.getService('TestManagement')
service.setTestSteps(wi.uri, test_steps.steps.TestStep)

#reload workitem so it's in sync
wi._reloadFromPolarion()
```

It did go farther but it failed when trying to set the test step:

> Exception has occurred: ValidationError
Missing element content (setTestSteps.testSteps.values.Text.content)
  File "C:\Hawk\ias_desktop_automation\spikes\testing_modify_steps.py", line 19, in <module>
    service.setTestSteps(wi.uri, test_steps.steps.TestStep)
zeep.exceptions.ValidationError: Missing element content (setTestSteps.testSteps.values.Text.content)

I feel that this validation error is incorrect because if you look at Test Service xml (https://almdemo.polarion.com/polarion/ws/services/TestManagementWebService?wsdl) it states that values under testSteps is supposed to be an ArrayOfText with the following definition:
```
complexType name="ArrayOfText">
<sequence>
<element maxOccurs="unbounded" minOccurs="0" name="Text" type="tns2:Text"/>
</sequence>
</complexType>
```
which means that "setTestSteps.testSteps.values.Text" is supposed to be an array and so content should be within an element in the Text.

Is this a bug in the python library or something further down in the SDK?

**Comment by jesper-raemaekers** [comment]
I think this might be the same cause as discovered in #100. In my case i got the same error as you did and the fix suggested there does seem to fix it so the example works again.

**Comment by vjacobjo** [comment]
Hi,
Thank you so much for replying back and giving all the suggestions. They were helpful.

Turns out, the error I was getting was due to the fact that I had a Text Object in my Test Steps where the content was "None" and it complained about that. Once I set that value to a string value ("N/A"), it started to work.

It also looks like that when you get test steps on a test case work item with no test steps, chances are that you would have text Items that have content set to "none". The user would have to know that if they are to reset the steps, that they must make sure all contents that were "None:" have to be set a value.

Maybe perhaps as an improvement, maybe have some sort of method that get test steps that will already ensure that content set to "None" will have at least an empty string character? "".

Thank you once again for maintaining this plugin. I find this really useful

**Comment by jesper-raemaekers** [comment]
Yeah, I did run into that issue as well. But I think i have all cases covered now. I've created tests and update the documentation in #102 so I hope that will work better for you. It should be released shortly.

**Comment by jesper-raemaekers** [comment]
This was released in V1.2.0 and I think this will cover your needs. Please close the issue when you tried these changes. Thanks!

**Comment by jesper-raemaekers** [comment]
> Couple of feedback points:
> 
>     1. Wow would you recommend modifying a step? I was expecting a new method similar to add and remove as follows:

Uh yes, Just didn't think of it :)  It will exist in the next release as 'updateTestStep' and will work as you would expect.

> Maybe this would be a future enhancement.
> 
>     2. When lets say I don't have the right amount of parameters specified for the addTestStep. The error message is very cryptic:
Good spot, looks like a mistake there. It will now also produce a readable error.

**Comment by vjacobjo** [comment]
Thank you for addressing these so quickly.

I can test it out for you and then close this ticket when I get the fix. When would this branch be merged and then the new version be published?

**Comment by jesper-raemaekers** [comment]
I'll push out a new release tonight :)

**Comment by vjacobjo** [comment]
Hi, I was able to get the latest version and then verify that:
1. the Error message for the add test step works as expected
2. The updateTestStep happy Path works as expected

One small issue:
I tried a case where I call the updateTestStep without the index. The error message is also cryptic here:

> Traceback (most recent call last):
  File "c:\Hawk\ias_desktop_automation\spikes\testing_modify_steps.py", line 29, in <module>
    wi.updateTestStep("New Modified Step", "New Modified step description", "Pass")
  File "C:\Program Files\Python39\lib\site-packages\polarion\workitem.py", line 627, in updateTestStep
    if index >= len(self._polarion_test_steps.steps.TestStep):
TypeError: '>=' not supported between instances of 'str' and 'int'

This could be easily resolved if you add an assert that index is an integer:
```
assert type(index) == int, "First argument should be an integer that specifies the the position of the test step to be updated."
```

On an unrelated question, am I allowed to create a branch to fix something in the codebase?
